pkg/certificateprovider/certificate_provider.go

```diff
@@ -12,5 +13,30 @@ type CertStoreConfig map[string]string
 // CertificateProvider is an interface that defines methods to be implemented by a each certificate provider
 type CertificateProvider interface {
 	// Returns an array of certificates based on certificate properties defined in attrib map
-	GetCertificatesContent(ctx context.Context, attrib map[string]string) ([]types.Certificate, error)
+	GetCertificates(ctx context.Context, attrib map[string]string) ([]*x509.Certificate, error)
+}
+
+// Decode PEM-encoded bytes into an x509.Certificate chain.
+func DecodeCertificates(value []byte) ([]*x509.Certificate, error) {
+	var certs []*x509.Certificate
+	block, rest := pem.Decode(value)
+	if block == nil && len(rest) > 0 {
+		return nil, fmt.Errorf("failed to decode pem block")
```

Comment:
just curious if we need to support cert in DER format in the future.

Reply:
From my side: not sure. I can't remember the last time I touched a DER cert. I can only speak for myself, but it's straightforward to convert formats w/ openssl or similar and doesn't feel unreasonable to expect of users.

One option might be to support PEM initially, and if there's enough usage/feedback for DER/other formats, add a new `format` field to the spec that defaults to PEM